mrp_subcontracted/models/purchase.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Akretion (http://www.akretion.com).
+# @author Raphaël Reverdy <raphael.reverdy@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# ajouter les fournisseurs sur les services
+
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    @api.multi
+    def _is_service_procurement(self):
+        """Ensure the order line is a service procurement.
+
+        It's tied to a procurement.
+        The procurement is tied to one manufacturing order
+        the product is a subcontracted_service
+
+        Multiple procurements or MOs not implemented
+        """
+        self.ensure_one()
+        is_service = False
+        has_procurement = False
+        has_mo = False
+        if self.procurement_ids:
+            has_procurement = (
+                self.procurement_ids.product_id == self.product_id)
+            has_mo = len(self.procurement_ids.production_id) == 1
```

Comment:
Je comprends pas trop cette partie. Je pense que tu n'as pas besoin de vérifier que self.procurement_ids == self.product_id, vu que c'est obligatoirement le cas. Simplement en faisant 'if self.procurement_ids', tu sais déjà que has_procurement est vrai.

2° Chose, je pense qu'il faut éviter le self.procurement_ids.product_id, car il va planter dans le cas d'une ligne avec 2 procurements. Pas sûr que ca puisse être possible avec nos produits de type service, etc... Mais la méthode is_service_procurement est appelée sur toutes les lignes. Donc si on a 1 ligne avec 2 produits stockables qui ont été mergés, on aura  2 procurement et odoo va planter.

Tu peux faire plutôt un truc comme
has_procurement = self.procurement_ids and len(self.procurement_ids) == 1 or False
if has_procurement:
    has_mo = len(self.procurement_ids.production_id) == 1
qu'en penses-tu?

Reply:
has_mo implique has_procurement, je vais donc virer has_procurement.
Je pense donc faire simplement : 
```py
        if is_service and self.procurement_ids:
            has_mo = len(self.procurement_ids.production_id) == 1
```
Et pour le cas où on a plusieurs procurements pour un service... et bien on avisera après :)